### DISCUSSION TITLE
PovertyMap Images

### DISCUSSION THREAD

**Discussion by niki-amini-naieni** [p-lambda/wilds#111] [Q&A] [answered]
Hello,

I know the PovertyMap dataset includes images with eight different channels. As a result, I am wondering which channels the authors plot in the following image:

<img width="849" alt="image" src="https://user-images.githubusercontent.com/61301874/158443524-747585cf-4b3f-4a41-860e-2ff162d63810.png">

(p. 15 of original paper (link here: https://arxiv.org/pdf/2012.07421.pdf))

Thanks!

**Comment by sangmichaelxie** [comment, marked as answer]
The plotted images are the RGB channels without mean and std normalization,
and then divided by the maximum pixel value of the entire image. Hope that
helps!

**Reply by niki-amini-naieni** [reply]
Are the RGB channels the first three (out of the eight) channels?

**Comment by sangmichaelxie** [comment]
Yes, the channel order is the order in which it appears in the MEAN and
STD:

BLUE, GREEN, RED, SWIR1, SWIR2, TEMP1, NIR, DMSP/VIIRS

**Reply by niki-amini-naieni** [reply]
Thank you. I see 9 means and standard deviations in the code shown below:
<img width="350" alt="image" src="https://user-images.githubusercontent.com/61301874/158664105-f6e4ed46-a64d-41fc-8056-698db00379ad.png">
How do you combine the last two channels to produce a total of eight channels and how do you incorporate the means and standard deviations of the last two channels into the construction of the eighth channel? Also, how do you justify using the means and standard deviations of all the images (including images in the test set) to normalize images in the training and validation sets?

**Reply by niki-amini-naieni** [reply]
I see that the nightlight data comes from two different sources and images from each source were normalized separately (i.e., each with a separate mean and standard deviation). I am including the following image and link for reference in case anyone else has the same question in the future.
<img width="690" alt="image" src="https://user-images.githubusercontent.com/61301874/158802841-4690ad4d-436e-4cd1-bae8-af8501043c82.png">
https://www.nature.com/articles/s41467-020-16185-w.pdf

**Comment by sangmichaelxie** [comment]
That's right, before 2012 the nightlights are from DMSP and for 2012 and
after it is from VIIRS. We just followed the normalization process from Yeh
et al. (https://www.nature.com/articles/s41467-020-16185-w). The means and
std shouldn't be test set statistics.

**Reply by niki-amini-naieni** [reply]
Thank you for the clarification! Perhaps I am misinterpreting the comment below. Could you please let me know what this comment implies?
<img width="600" alt="image" src="https://user-images.githubusercontent.com/61301874/158973191-04104728-074e-4602-bf60-f1367c7c5701.png">
(https://github.com/p-lambda/wilds/blob/main/dataset_preprocessing/poverty/dataset_constants.py)

**Reply by sangmichaelxie** [reply]
Actually, you're right - again, we are following Yeh et al., but I believe the reason this was done is because we test on 5 different folds of the data, and 1) the average evaluation metric over folds is already a function of all the folds, 2) convenience, but I agree that perhaps a better way to do it is to compute a fold-specific mean/std using each training set.

**Reply by niki-amini-naieni** [reply]
Thank you for the confirmation! I completely understand (:
